Meshtastic/Helpers/BLEManager.swift

```diff
@@ -866,7 +866,10 @@ class BLEManager: NSObject, CBPeripheralDelegate, MqttClientProxyManagerDelegate
 							let traceRouteHop = TraceRouteHopEntity(context: context)
 							traceRouteHop.time = Date()
 							if routingMessage.snrTowards.count >= index + 1 {
-								traceRouteHop.snr = Float(routingMessage.snrTowards[index] / 4)
+								traceRouteHop.snr = Float(routingMessage.snrTowards[index]) / 4
+							} else {
+								// If no snr in route, try last snr from node
+								traceRouteHop.snr = hopNode?.snr ?? 0
```

Comment:
Why we are pulling readings from the nodeDb here?

Isn't this going to possibly be `SNR(selfNode <- currentHopNode)` when we actually want, `SNR(currentHopNode <- previousHopNode)`?

Reply:
I've removed the use of the node snr here. If the routing message doesn't have a valid snr we make it unknown to reflect what really happened during the traceroute.